gst/datarepo/gstdatareposink.c

```diff
@@ -300,29 +307,69 @@ static GstFlowReturn
 gst_data_repo_sink_write_flexible_tensors (GstDataRepoSink * sink,
     GstBuffer * buffer)
 {
+  guint num_tensors, i;
+  gsize write_size = 0, total_write = 0, tensor_size;
   GstMapInfo info;
-  gsize to_write = 0;
+  GstMemory *mem;
+  GstTensorMetaInfo meta;
 
   g_return_val_if_fail (sink != NULL, GST_FLOW_ERROR);
   g_return_val_if_fail (buffer != NULL, GST_FLOW_ERROR);
   g_return_val_if_fail (sink->fd != 0, GST_FLOW_ERROR);
   g_return_val_if_fail (sink->json_object != NULL, GST_FLOW_ERROR);
-  g_return_val_if_fail (sink->json_offset_array != NULL, GST_FLOW_ERROR);
+  g_return_val_if_fail (sink->buffer_offset_array != NULL, GST_FLOW_ERROR);
+  g_return_val_if_fail (sink->tensor_size_array != NULL, GST_FLOW_ERROR);
 
   GST_OBJECT_LOCK (sink);
-  gst_buffer_map (buffer, &info, GST_MAP_READ);
 
-  to_write = info.size;
+  num_tensors = gst_buffer_n_tensor (buffer);
+  GST_INFO_OBJECT (sink, "num_tensors: %u", num_tensors);
 
-  gst_buffer_unmap (buffer, &info);
-  GST_OBJECT_UNLOCK (sink);
+  for (i = 0; i < num_tensors; i++) {
+    mem = gst_tensor_buffer_get_nth_memory (buffer, i);
```

Comment:
You should unref the mem (gst_memory_unref (mem);) when using gst_tensor_buffer_get_nth_memory().

Reply:
Thank you.